### DISCUSSION TITLE
Calendar block: add option to pick a category of posts

### DISCUSSION THREAD

**Discussion by ghost** [WordPress/gutenberg#50652] [Ideas]
I was looking for a simple way to use the calendar block as a calendar for events. Currently the calendar block has no option at all and will only show dates that have posts, all posts. Adding category selection to this block would allow to use this block for much more purposes, like in my case, only dates that have posts of a specific gallery. I would use the category 'Agenda' for all kinds of events, like meetings, lectures, parties, whatever. Now it is only possible by using an extra plugin with way too many options, thus bloating, while many situations could be tackled by adding category selection to the calendar block.

Of course there are more options that could be added, like selecting two categories and pick a different color for each category, but just to be able to select one category would be very nice.

**Comment by t-hamano** [comment]
Hi @kneops,

The calendar block is [rendered server-side using the `get_calendar()` function](https://github.com/WordPress/gutenberg/blob/4deac6282413e8cbfc262325912a6dc6187bb03b/packages/block-library/src/calendar/index.php#L63). However, this function is [hardcoded SQL](https://github.com/WordPress/wordpress-develop/blob/0b10d2fc06d1a73f22c4ff506ed8d2f1827a8cb6/src/wp-includes/general-template.php#L2345-L2351) to retrieve posts and does not appear to support filtering by custom post type, category, or taxonomy.

This makes it difficult to support on the block side and may be better to ticket WordPress core as an enhancemnent. From my research, it appears that [there is a ticket regarding a new filter to refine by custom post type](https://core.trac.wordpress.org/ticket/34093).

**Comment by ghost** [comment]
Thanks for all the info and the link! Very helpful.
